crates/sui-indexer-alt-jsonrpc/src/api/write.rs

```diff
@@ -0,0 +1,90 @@
+use sui_json_rpc_types::{
+    DryRunTransactionBlockResponse, SuiTransactionBlockResponse, SuiTransactionBlockResponseOptions,
+};
+use sui_open_rpc::Module;
+use sui_open_rpc_macros::open_rpc;
+use sui_types::quorum_driver_types::ExecuteTransactionRequestType;
+
+use crate::error::internal_error_object;
+
+use super::rpc_module::RpcModule;
+
+#[open_rpc(namespace = "sui", tag = "Write API")]
+#[rpc(server, client, namespace = "sui")]
+pub trait WriteApi {
+    /// Execute the transaction and wait for results if desired.
+    /// Request types:
+    /// 1. WaitForEffectsCert: waits for TransactionEffectsCert and then return to client.
+    ///     This mode is a proxy for transaction finality.
+    /// 2. WaitForLocalExecution: waits for TransactionEffectsCert and make sure the node
+    ///     executed the transaction locally before returning the client. The local execution
+    ///     makes sure this node is aware of this transaction when client fires subsequent queries.
+    ///     However if the node fails to execute the transaction locally in a timely manner,
+    ///     a bool type in the response is set to false to indicated the case.
+    /// request_type is default to be `WaitForEffectsCert` unless options.show_events or options.show_effects is true
+    #[method(name = "executeTransactionBlock")]
+    async fn execute_transaction_block(
+        &self,
+        /// BCS serialized transaction data bytes without its type tag, as base-64 encoded string.
+        tx_bytes: Base64,
+        /// A list of signatures (`flag || signature || pubkey` bytes, as base-64 encoded string). Signature is committed to the intent message of the transaction data, as base-64 encoded string.
+        signatures: Vec<Base64>,
+        /// options for specifying the content to be returned
+        opts: Option<SuiTransactionBlockResponseOptions>,
+        /// The request type, derived from `SuiTransactionBlockResponseOptions` if None
+        request_type: Option<ExecuteTransactionRequestType>,
+    ) -> RpcResult<SuiTransactionBlockResponse>;
+
+    /// Return transaction execution effects including the gas cost summary,
+    /// while the effects are not committed to the chain.
+    #[method(name = "dryRunTransactionBlock")]
+    async fn dry_run_transaction_block(
+        &self,
+        tx_bytes: Base64,
+    ) -> RpcResult<DryRunTransactionBlockResponse>;
+}
+
+pub(crate) struct Write(pub HttpClient);
+
+#[async_trait::async_trait]
+impl WriteApiServer for Write {
+    async fn execute_transaction_block(
+        &self,
+        tx_bytes: Base64,
+        signatures: Vec<Base64>,
+        opts: Option<SuiTransactionBlockResponseOptions>,
+        request_type: Option<ExecuteTransactionRequestType>,
+    ) -> RpcResult<SuiTransactionBlockResponse> {
+        Ok(self
+            .0
+            .execute_transaction_block(tx_bytes, signatures, opts, request_type)
+            .await
+            .map_err(internal_error_object)?)
```

Comment:
IIUC, this means that any error, including user errors, transient errors, etc from the fullnode will turn into internal errors here. That seems wrong -- we should be passing the errors on transparently.